rootston/seat.c

```diff
@@ -14,6 +15,7 @@
 static void handle_keyboard_key(struct wl_listener *listener, void *data) {
 	struct roots_keyboard *keyboard =
 		wl_container_of(listener, keyboard, keyboard_key);
+	wl_signal_emit(&keyboard->seat->seat->events.activity_notify, NULL);
```

Comment:
I don't like emitting the raw event on the seat signal from rootston. Having the signal on the seat just for the one protocol also doesn't seem right to me. What about notifying the wlr-idle itself from here? Sort of like `wlr_idle_notify_activity(seat)` which then handles the event on its own.

Reply:
+1. `wlr_idle_notify_activity(idle, seat)` could iterate over all clients and update them accordingly.

Reply:
that means that I should have wlr_idle accessible here in rootston seat? this will lead to removing wlr_seat form idle, but that's not a problem.

Reply:
You can access idle from rootston via `desktop->idle`.